### DISCUSSION TITLE
Adding a title to a code block requires a space separator since some point release of 9.5

### DISCUSSION THREAD

**Discussion by polarathene** [squidfunk/mkdocs-material#7687] [General]
**Resolved:** Temporary regression introduced Sep 29, resolved shortly after. For some reason I thought I had seen a recent build with the same issue but was mistaken.

---

**Docs:** Code block - [Adding a title](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title)

I'd have opened a bug report, but lack the time to fulfill all the requirements. Potentially a regression bug or perhaps intentional, sharing here should it be of any benefit.

---

At some point in the past 10 months with the `9.5` series there was a regression with the feature.

- A code block without a syntax hint, only a `title` attribute directly after the three backticks no longer renders the title, breaking the code block rendering.
- This is [remedied by inserting a space delimiter](https://github.com/docker-mailserver/docker-mailserver/pull/4258#issuecomment-2466876488) between the backticks and the title attribute.

---

Working:

> ![image](https://github.com/user-attachments/assets/a1d0c1c4-afca-4ea1-9e9a-ef53128bc05f)

Broken / Regression:

> ![image](https://github.com/user-attachments/assets/fc850462-4728-4181-9507-749fc18386bc)

**Comment by squidfunk** [comment]
> I'd have opened a bug report, but lack the time to fulfill all the requirements. Potentially a regression bug or perhaps intentional, sharing here should it be of any benefit.

Github Discussions it not intended for bug reports, [as is mentioned in our guidelines](https://squidfunk.github.io/mkdocs-material/contributing/#before-asking-a-question). I kindly ask you to fulfill the requirements for reporting a bug, or we're unable to look into it.

**Reply by facelessuser** [reply]
I agree that proper behavior should be followed according to the guidelines of the repository, but since I know this is _not_ controlled by Material and controlled by Pymdown Extensions, and since I also know this is working in the latest version of Pymdown Extensions, my advice would be to make sure you are on the latest version:

````py
import markdown
import pymdownx

print(f'Pymdown Extensions Version: {pymdownx.__version__}')
print(f'Markdown Version: {markdown.__version__}')
print('-----')

MD = """
``` title="Title"
Some text
```

```title="Title"
Some text
```
"""

extensions = ["pymdownx.superfences"]
extension_configs = {}

print(markdown.markdown(MD, extensions=extensions, extension_configs=extension_configs))
````

```
Pymdown Extensions Version: 10.12
Markdown Version: 3.7
-----
<div class="highlight"><span class="filename">Title</span><pre><span></span><code>Some text
</code></pre></div>
<div class="highlight"><span class="filename">Title</span><pre><span></span><code>Some text
</code></pre></div>
```

If you are still having issues after that, please file an issue over at https://github.com/facelessuser/pymdown-extensions/issues. But do take the time to fill out a proper issue with examples or we will also not be able to look at it.

**Reply by polarathene** [reply]
It's unlikely the regression will bother anyone, I'm sure your both very busy yourselves thus I wouldn't worry about tracking it down.

**TL;DR:** Assuming it doesn't reproduce with `mkdocs-material`, nor a difference due to the Docker image `squidfunk/mkdocs-material:9.5`, then some `9.5.x` release since June became incompatible with something on my end.

---

> or we're unable to look into it.

I understand, but as stated the expectations of raising a bug report with the necessary info isn't something I have time for right now for this.

It's a fairly minor regression, simply adding a space delimiter as mentioned resolves the rendering issue, so I doubt anyone will be that concerned by it. I figured it was still worth making other users aware of, should the run into it too, I'm not fussed about it being addressed.

---

> since I also know this is working in the latest version of Pymdown Extensions, my advice would be to make sure you are on the latest version

Our CI hasn't changed the way we build the image for over 10 months, nor was the page displayed modified in that time that would affect the rendering differently:

```bash
docker run \
  --rm \
  --user "$(id -u):$(id -g)" \
  --volume "${PWD}:/docs" \
  --name "build-docs" \
  squidfunk/mkdocs-material:9.5 build --strict
```

So while I'm sure you know what you're talking about, either something in the published official `mkdocs-material` image, or part of the 9.5 point release series, a regression did occur.

Without taking the time to investigate further and produce a minimal reproduction, I won't know where the regression was introduced. It's possible that the docs project built has some incompatibility since a `9.5.x` mkdocs-material release, all I know is something changed external of the affected docs project that introduced the regression.

Our [last release prior to the regression was in June](https://github.com/docker-mailserver/docker-mailserver/releases/tag/v14.0.0), so the change came from upstream somewhere since then.

---

> But do take the time to fill out a proper issue with examples or we will also not be able to look at it.

I normally would, I'm just unable to justify the time for this with my current workload.

**Reply by facelessuser** [reply]
@polarathene What I'm saying is that any issue would be with Pymdown Extensions, the SuperFences extension in that library specifically. IIRC, only recently do we allow no spaces before options when a language is not specified. If this worked at any time before very recent revisions, it was by accident. This I believe was done in https://github.com/facelessuser/pymdown-extensions/releases/tag/10.11, but there was a bug and it was fixed in https://github.com/facelessuser/pymdown-extensions/releases/tag/10.11.1.

To be honest, if the docker image is locked to version of Pymdown Extensions in the 10.11.X series, I would advise it be at least https://github.com/facelessuser/pymdown-extensions/releases/tag/10.11.2, but with 10.12.X being available, I'd probably recommend that.

**Reply by polarathene** [reply]
> but with 10.12.X being available, I'd probably recommend that.

Latest image has `10.12` installed according to `pip list`.

> IIRC, only recently do we allow no spaces before options when a language is not specified. If this worked at any time before very recent revisions, it was by accident.

Ah alright. I can see that it works without needing the space workaround.

Looks like the current pre-release docs for the project were last built/published on Sep 29, which was when a new release of mkdocs-material was made and that also bumped to `pymdown-extensions 10.11`:

```console
$ docker run --rm -it --entrypoint /usr/local/bin/pip squidfunk/mkdocs-material:9.5.39 list | grep pymdown-extensions
pymdown-extensions         10.11
```

So it would have been affected by that bug from `10.11`, which you quickly resolved with `10.11.1`.

Apologies for the noise, I thought a preview build had been done more recently but those were also last built on Sep 29.

**Reply by squidfunk** [reply]
We depend on quite a few substantial downstream projects, and we haven't pinned dependencies, so anything that gets released in those dependencies as part of a semver-compatible minor release gets picked up by Material for MkDocs. It's always a good idea to isolate which dependency it is, most easily by using another theme than Material for MkDocs. By the way, we decided to use semver, [which also got us into trouble with some very strong opinions](https://github.com/squidfunk/mkdocs-material/issues/7076).
